core/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -258,8 +258,12 @@ private static Fields generateTermVectors(IndexShard indexShard, Map<String, Obj
         for (Map.Entry<String, Collection<Object>> entry : values.entrySet()) {
             String field = entry.getKey();
             Analyzer analyzer = getAnalyzerAtField(indexShard, field, perFieldAnalyzer);
-            for (Object text : entry.getValue()) {
-                index.addField(field, text.toString(), analyzer);
+            if (entry.getValue() instanceof List) {
+                for (Object text : entry.getValue()) {
+                    index.addField(field, text.toString(), analyzer);
```

Comment:
Some mean person is going to write `"fields": ["bleh", null, "blah"]` and complain about the weird exception at this line. Maybe just throw an `IllegalArgumentException` here.